tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,767 @@
+        // Checking pre-conditions
+        if (self.key_alias.is_some() && self.regions.is_none())
+            || (self.key_alias.is_none() && self.regions.is_some())
+        {
+            error::KeyConfig.fail()?;
+        }
+        if self.available_keys.is_some() {
+            //available keys is either empty or populated with current values
+            new_available_keys = self
+                .available_keys
+                .as_ref()
+                .context(error::MissingConfig {
+                    missing: "available_keys",
+                })?
+                .clone();
+        }
+
+        // Generating new keys
+        // NOTE: using unwrap here because regions and alias existence is checked above
+        for region in self.regions.as_ref().unwrap().iter() {
+            let stack_name = format!("TUF-KMS-{}", self.key_alias.as_ref().unwrap());
+            let cfn_client = CloudFormationClient::new(
+                Region::from_str(region).context(error::ParseRegion { what: region })?,
+            );
+            let cfn_filepath = format!(
+                "{}/infrasys/cloudformation-templates/kms_key_setup.yml",
+                getenv("BUILDSYS_TOOLS_DIR")?
+            );
+            let cfn_template = fs::read_to_string(&cfn_filepath)
+                .context(error::FileRead { path: cfn_filepath })?;
+
+            let stack_result = cfn_client
+                .create_stack(CreateStackInput {
+                    parameters: Some(vec![get_parameter(
+                        String::from("Alias"),
+                        self.key_alias.as_ref().unwrap().to_string(),
+                    )]),
+                    stack_name: stack_name.clone(),
+                    template_body: Some(cfn_template.clone()),
+                    ..Default::default()
+                })
+                .await
+                .context(error::CreateStack {
+                    stack_name: &stack_name,
+                    region,
+                })?;
+
+            let stack_arn = stack_result
+                .clone()
+                .stack_id
+                .context(error::ParseResponse {
+                    what: "stack_id",
+                    resource_name: &stack_name,
+                })?;
+
+            let mut stack_outputs = cfn_client
+                .describe_stacks(DescribeStacksInput {
+                    next_token: None,
+                    stack_name: Some(stack_name.clone()),
+                })
+                .await
+                .context(error::DescribeStack {
+                    stack_name: &stack_name,
+                    region,
+                })?
+                .stacks
+                .context(error::ParseResponse {
+                    what: "stacks",
+                    resource_name: &stack_name,
+                })?[0] //QUESTION: This error handling okay?
```

Comment:
Yeah, this is fine. You can make it more clear by just assigning to more intermediary variables, but some people might not like that.